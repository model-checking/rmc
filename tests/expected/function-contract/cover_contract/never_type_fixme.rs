// Copyright Kani Contributors
// SPDX-License-Identifier: Apache-2.0 OR MIT
// kani-flags: -Zfunction-contracts

// Unreachable postcondition because the function never returns.
// Kani should fail verification because the postcondition is unreachable,
// but currently doesn't generate the postcondition check at all
// (although verification still fails because of the unwinding error).
// We may need special never type detection for this case.

#![feature(never_type)]

#[kani::requires(true)]
#[kani::ensures(|result: &!| true)]
fn never_return() -> ! {
    let x = 7;
    loop {}
}

// Unreachable postcondition because the function never returns
#[kani::proof_for_contract(never_return)]
#[kani::unwind(5)]
fn prove_never_return() {
    never_return();
}
