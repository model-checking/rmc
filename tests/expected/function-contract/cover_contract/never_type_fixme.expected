contract_cover.2\
	 - Status: FAILURE\
	 - Description: "The contract's postcondition is reachable."

unwind.0\
    - Status: FAILURE\
    - Description: "unwinding assertion loop 0"

VERIFICATION:- FAILED
